test/libsolidity/semanticTests/various/code_length_contract_member.sol

```diff
@@ -0,0 +1,17 @@
+// Test to see if type.code.length does extcodesize(type) only when type is an address.
+struct S {
+    bytes32 code;
+    bytes32 another;
+}
+
+contract C {
+    S s;
+
+    function f() public returns (uint, uint, bool) {
+        return (s.code.length, s.another.length, address(this).code.length > 150);
```

Comment:
Can you add another test which uses
```
address a = address(this);
a.code.length > 150
```

The reason: if we would later add a case to replace `address(this).code.length` with `CODESIZE`, at least the intermediate variable would make that lookup less likely in the old codegen, so it would remain on `EXTCODESIZE.`

Probably the real solution would be adding some compilation test which ensures there is an `EXTCODESIZE` generated.

Reply:
There is `test/libsolidity/semanticTests/various/address_code.sol`, that has `address(0).code.length`.

I'll probably add a command line test which outputs the asm.

Reply:
Ah, `address(0)` should be good enough. So perhaps then can you just adjust the tests so that we have `address(0)` and `address(this)` in both the constructor and outside of it?

That should cover all these potential cases. In the constructor we should never use the non-ext versions for `address(this)`.

Reply:
I added a cmdlinetest.